ctapipe/image/geometry_converter.py

```diff
@@ -399,3 +483,217 @@ def convert_geometry_back(geom, signal, key, add_rot=0):
+        # indices
+        for i, row in enumerate(square_mask):
+            for j, val in enumerate(row):
+                if val is True:
+                    ids.append((i, j))
+
+        # the area of the pixels (note that this is still a deformed
+        # image)
+        pix_area = np.ones_like(grid_x) \
+            * (x_edges[1] - x_edges[0]) * (y_edges[1] - y_edges[0])
+
+        # creating a new geometry object with the attributes we just determined
+        new_geom = CameraGeometry(
+            cam_id=geom.cam_id + "_rect",
+            pix_id=ids,  # this is a list of all the valid coordinate pairs now
+            pix_x=grid_x * u.m,
+            pix_y=grid_y * u.m,
+            pix_area=pix_area * u.m ** 2,
+            neighbors=geom.neighbors,
+            pix_type='rectangular', apply_derotation=False)
+
+        # storing the pixel mask for later use
+        new_geom.mask = square_mask
+
+        # create a transfer map by enumerating all pixel positions in a 2D histogram
+        hex_to_rect_map = np.histogramdd([rot_y, rot_x],
+                                         bins=(y_edges, x_edges),
+                                         weights=np.arange(len(signal)))[0].astype(int)
+        # bins that do not correspond to the original image get an entry of `-1`
+        hex_to_rect_map[~square_mask] = -1
+
+        if signal.ndim > 1:
+            long_map = []
+            for i in range(signal.shape[-1]):
+                tmp_map = hex_to_rect_map + i * (np.max(hex_to_rect_map) + 1)
+                tmp_map[~square_mask] = -1
+                long_map.append(tmp_map)
+            hex_to_rect_map = np.array(long_map)
+
+        if key is not None:
+            # if a key is given, store the essential objects in a buffer
+            rot_buffer[key] = (geom, new_geom, hex_to_rect_map)
+
+    # done `if key in rot_buffer`
+
+    # create the rotated rectangular image by applying `hex_to_rect_map` to the flat,
+    # extended input image
+    # `input_img_ext` is the flattened input image extended by one entry that contains NaN
+    # since `hex_to_rect_map` contains `-1` for "fake" pixels, it maps this extra NaN
+    # value at the last array position to any bin that does not correspond to a pixel of
+    # the original image
+    input_img_ext = np.full(np.prod(signal.shape) + 1, np.nan)
+
+    # the way the map is produced, it has the time dimension as axis=0;
+    # but `signal` has it as axis=-1, so we need to roll the axes back and forth a bit.
+    # if there is no time dimension, `signal` is a 1d array and `rollaxis` has no effect.
+    input_img_ext[:-1] = np.rollaxis(signal, axis=-1, start=0).ravel()
+
+    # now apply the transfer map
+    rot_img = input_img_ext[hex_to_rect_map]
+
+    # if there is a time dimension, roll the time axis back to the last position
+    try:
+        rot_img = np.rollaxis(rot_img, 0, 3)
+    except np.core._internal.AxisError:
```

Comment:
This fails for me on the latest version of NumPy (you shouldn't used any module marked with an underscore, as that is private and subject to API change).

Reply:
it seems it now throws a `ValueError` (which is the base class of the older `AxisError` anyhow, so catching a `ValueError` should work in both cases)

Reply:
I see... For me it was throwing this error when it wasn't working so I just caught it. I will change the catch to simple ValueError